src/main/java/com/SAS/LeagueManagement/LeagueManagementController.java

```diff
@@ -304,11 +303,9 @@ public User addNewReferee(List<String> details, User user) {
                 String pass = details.get(1);
                 String fullName = details.get(2);
                 int level = Integer.parseInt(details.get(3));
-                Referee referee = new Referee(user, fullName);
-                referees.add(referee);
-                if (LeagueCRUD.addReferee(referee)) {
+                if (LeagueCRUD.addReferee(UsersCRUD.getUserIdByUserName(userName),level)) {
```

Comment:
Calling LeagueCRUD.addReferee is good because you should create a record in referee's table, but you should also call userController.createUser with his details because this was you do not create a user for him and you should.. Otherwise UsersCRUD.getUserIdByUserName(userName) will deferentially return -1... 
Add userController.createUser(username, pass, fullname ....) before this